### PR TITLE
Re/feat/extension workspace

### PR DESCRIPTION
**Pull request by r3ym3l**
## What does this PR do?
Chrome context menu displays workspace from database. Highlight and save text is saved on to the workspace **texts** array and the workspace ID is also attached to the saved text.

### Screenshots (optional)
image

### Steps to test/review (optional)
- start back-end (front-end optional)
- build the extension and load into chrome browser
- highlight and save while checking the database to see changes: 
    - workspace ID where text was saved is included
    - new text is pushed on to the workspace it was saved in

## Checklist before merging
- [x] Applied options on the right:
      - Reviewers
      - Assignee
      - Labels
      - Project
      - Milestone
      - Linked Issues
- [x] All linter rules passed
- [x] Auto-build success

#96

### CODE REVIEW

File: back-end/persistence/models/workspaces.model.js [hunk 1]
```diff
@@ -8,27 +8,25 @@ const workspaceSchema = new Schema({
         required: true
     },
     owner: {
-        type: String,
-        required: true
+        type: String
```
Comment (mariellemanlulu) [inline back-end/persistence/models/workspaces.model.js:11]: required?

File: back-end/persistence/models/workspaces.model.js [hunk 2]
```diff
@@ -8,27 +8,25 @@ const workspaceSchema = new Schema({
...
     },
     collaborators: {
```
Comment (mariellemanlulu) [inline back-end/persistence/models/workspaces.model.js:13]: by default maybe add the owner here to?

File: front-end/public/background.js [hunk 3]
```diff
@@ -10,16 +9,16 @@ try {
...
     contexts: ["selection"],
   };
 
-  fetch(`${serverAddr}/accounts`)
-    .then((r) => r.text())
-    .then((result) => {
-    // Result now contains the response text, do what you want...
-      console.log("The fetch has been made. " + "The result is: " + result.JSON);
-    });
+  // fetch(`${serverAddr}/accounts`)
```
Comment (mariellemanlulu) [inline front-end/public/background.js:16]: we can remove this if we don't need it anymore

File: front-end/public/background.js [hunk 4]
```diff
@@ -31,29 +30,60 @@ try {
...
-    
-    if (helper.workspaceExists(clickData.menuItemId)) {
-
-      const notificationTextFormatted = helper.formatText(clickData.selectionText);
-
-      helper.createNotification(notificationTextFormatted);
-
-      helper.saveTextToDb(clickData.selectionText, clickData.pageUrl);
-    }
+    fetch(`${serverAddr}/workspaces`)
```
Comment (mariellemanlulu) [inline front-end/public/background.js:53]: we should break this down into multiple function? it's getting bigger.

File: front-end/src/organisms/Sidebar.js [hunk 5]
```diff
@@ -9,6 +9,47 @@ import "./Sidebar.css";
 
 function Sidebar() {
 
+  // FAKE DB FOR UI
```
Comment (mariellemanlulu) [inline front-end/src/organisms/Sidebar.js:12]: if we don't need this, we can remove it.

**Review by mariellemanlulu — CHANGES_REQUESTED** [review]

File: front-end/public/background.js (hunk 4 above)
Comment (mariellemanlulu) [reply front-end/public/background.js:53]: we can leave this in another branch if we need this branch merged right away

**Review by deleonkf — CHANGES_REQUESTED** [review]

**Review by deleonkf — COMMENTED** [review]
ty for fixing the tests! Might have missed committing the changes for `background.js` and `helper.js`

**Review by deleonkf — APPROVED** [review]

**Review by mariellemanlulu — APPROVED** [review]

**Pull request merged**